src/Data/Text/AhoCorasick/Searcher.hs

```diff
@@ -152,5 +154,35 @@ containsAny !searcher !text =
     -- On the first match, return True immediately.
     f _acc _match = Aho.Done True
   in case caseSensitivity searcher of
-    CaseSensitive  -> Aho.runText False f (automaton searcher) text
-    IgnoreCase      -> Aho.runLower False f (automaton searcher) text
+    CaseSensitive -> Aho.runText False f (automaton searcher) text
+    IgnoreCase   -> Aho.runLower False f (automaton searcher) text
+
+-- | Build a 'Searcher' that returns the needle's index in the needle list when it matches.
+buildNeedleIdSearcher :: CaseSensitivity -> [Text] -> Searcher Int
+buildNeedleIdSearcher !case_ !ns =
+  buildWithValues case_ $ zip ns [0.. length ns - 1]
```

Comment:
You can omit the upper limit of the index list (i.e. just `[0..]`), the output of zip is already clamped to the shorter of the two input lists. Saves you one pass over `ns`.

Reply:
True, it also makes for nicer code!